plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -280,6 +316,53 @@ def __init__(self, module, connection_header):
         self.restheaders = connection_header
         self.http_agent = self.module.params.get('http_agent')
 
+    def _request(self, url, method, data=None):
+        def make_request_catching_401():
+            try:
+                return open_url(url, method=method, data=data,
+                                http_agent=self.http_agent, headers=self.restheaders,
+                                timeout=self.connection_timeout,
+                                validate_certs=self.validate_certs)
+            except HTTPError as e:
+                if e.code != 401:
+                    raise e
+                return e
```

Comment:
This still bugs me on principle - exceptions are meant to be raised not returned :-) - but let's not hold this PR back because of it. I'll make a note to myself to return to this later and see if I can make it any better.